packages/salesforcedx-vscode-core/test/jest/context/contextVariables.test.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { ConfigUtil } from '@salesforce/salesforcedx-utils-vscode';
+import * as vscode from 'vscode';
+import { setIsScratchOrg } from '../../../src/context/contextVariables';
+import { OrgAuthInfo } from '../../../src/util';
+
+describe('contextVariables', () => {
+  describe('setIsScratchOrg', () => {
+    let getUsernameMock: jest.SpyInstance;
+    let isAScratchOrgMock: jest.SpyInstance;
+    let executeCommandSpy: jest.SpyInstance;
+
+    beforeEach(() => {
+      getUsernameMock = jest
+        .spyOn(ConfigUtil, 'getUsername')
+        .mockResolvedValue('test@test.com');
+
+      isAScratchOrgMock = jest.spyOn(OrgAuthInfo, 'isAScratchOrg');
+
+      executeCommandSpy = jest.spyOn(vscode.commands, 'executeCommand');
+    });
```

Comment:
With Jest, if I remember correctly the mocks and spies should be cleared after each test:
```
afterEach(() => {
  jest.clearAllMocks();
}
```

Reply:
Jest has some cool config options, so this is already done via a jest config setting in the VSCE repo.

Each package contains a jest.config.js file, which imports from the top-level jest.base.config file:
![2023-04-06_10-33-36](https://user-images.githubusercontent.com/46458081/230454134-8e8624ec-42a2-4a92-87e9-0d5acfcef9f0.png)


In that file the [resetMocks option](https://jestjs.io/docs/configuration#clearmocks-boolean) is set to true, which makes jest call clearAllMocks before each test:
![2023-04-06_10-33-52](https://user-images.githubusercontent.com/46458081/230454247-29997a45-641b-4339-9dec-59d3ecd7aa80.png)